src/Tasks/GetSDKReferenceFiles.cs

```diff
@@ -1246,63 +1221,105 @@ public SdkReferenceInfo(string fusionName, string imageRuntime, bool isWinMD, bo
-                using (var fs = new FileStream(cacheFile, FileMode.Open))
+                TranslateConcurrentDictionary<SdkReferenceInfo>(translator, ref pathToReferenceMetadata, (ITranslator t, ref SdkReferenceInfo info) =>
                 {
-                    var formatter = new BinaryFormatter();
-                    var info = (SDKInfo)formatter.Deserialize(fs);
+                    info ??= new SdkReferenceInfo(null, null, false, false);
+                    string fusionName = info.FusionName;
+                    string imageRuntime = info.ImageRuntime;
+                    bool isManagedWinmd = info.IsManagedWinmd;
+                    bool isWinmd = info.IsWinMD;
+                    t.Translate(ref fusionName);
+                    t.Translate(ref imageRuntime);
+                    t.Translate(ref isManagedWinmd);
+                    t.Translate(ref isWinmd);
+                    info.FusionName = fusionName;
+                    info.ImageRuntime = imageRuntime;
+                    info.IsManagedWinmd = isManagedWinmd;
+                    info.IsWinMD = isWinmd;
+                });
 
-                    // If the serialization versions don't match, don't use the cache
-                    if (info != null && info._serializedVersion != CurrentSerializationVersion)
-                    {
-                        return null;
-                    }
+                TranslateConcurrentDictionary<List<string>>(translator, ref directoryToFileList, (ITranslator t, ref List<string> fileList) =>
+                {
+                    t.Translate(ref fileList, (ITranslator t, ref string str) => { t.Translate(ref str); });
+                });
+
+                translator.Translate(ref hash);
+            }
+        }
 
-                    return info;
+        private static void TranslateConcurrentDictionary<T>(ITranslator translator, ref ConcurrentDictionary<string, T> dictionary, ObjectTranslator<T> objTranslator)
+        {
+            int count = dictionary.Count;
+            translator.Translate(ref count);
+            string[] keys = dictionary.Keys.ToArray();
+            for (int i = 0; i < count; i++)
+            {
+                if (i < keys.Length)
+                {
+                    translator.Translate(ref keys[i]);
+                    T value = dictionary[keys[i]];
+                    objTranslator(translator, ref value);
                 }
+                else
+                {
+                    string key = null;
+                    translator.Translate(ref key);
+                    T value = default;
+                    objTranslator(translator, ref value);
+                    dictionary[key] = value;
+                }                
             }
         }
 
         /// <summary>
         /// This class represents the context information used by the background cache serialization thread.
         /// </summary>
-        private class SaveContext
+        internal class SaveContext
```

Comment:
Why did this need to change?

Reply:
I had to fiddle with a lot of these to make them accessible to the test class. Then I downgraded a couple because properties were more visible than their containing class, which isn't really good style, but the latter wasn't necessary. As long as something prevents it from being public/protected, I don't think it particularly matters.

Reply:
Ah, I hadn't reviewed the test so I didn't see _any_ need for it.